psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -2,130 +2,140 @@
-
-        //connect to host server
-        DataInputStream fromHost;
-        DataOutputStream toHost;
-        while (true) {
-            try {
-                System.out.println("Attempting to connect to localhost:9999");
-                Socket socket = new Socket();
-                socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), 9999), 5000);
-                fromHost = new DataInputStream(socket.getInputStream());
-                toHost = new DataOutputStream(socket.getOutputStream());
-                break;
-            } catch (Exception e) {
-                System.err.println("Retrying: " + e.getMessage());
-                Thread.sleep(2000);
+    enum Role {
+        MERCHANT,
+        SERVICE_PROVIDER
+    }
+
+    @Override
+    public Void call() throws Exception {
+        //a new private key is generated. Enclave Client is created using this private key and constraint.
+        //a corresponding public key will be used by the enclave to encrypt data to be sent to this client
+        EnclaveClient enclaveClient = new EnclaveClient(constraint);
+
+        try (WebEnclaveTransport transport = new WebEnclaveTransport(url);
+             EnclaveClient client = enclaveClient)
+        {
+            //retrieve the enclaveInstanceInfo object, i.e. the remote attestation object and verify it against the
+            //constraint
+            client.start(transport);
+
+            //collect merchants and service providers credit card numbers list
+            InputData inputData = getInputData();
+
+            byte[] requestMailBody = serializeMessage(inputData).getBuffer();
+
+            //client will send its credit card list to enclave and wait for other client to send their list
+            EnclaveMail responseMail = client.sendMail(requestMailBody);
+
+            //responseMail is null till enclave doesn't reply back to the client
+            if(responseMail == null) {
+                do {
+                    Thread.sleep(5000);
+                    //poll for reply to enclave
+                    responseMail = enclaveClient.pollMail();
+                } while (responseMail==null);
             }
+            System.out.println("Ad Conversion Rate is : " + new String(responseMail.getBodyAsBytes()));
         }
+        return null;
+    }
 
-        //take inputs from user
+    private InputData getInputData() {
         InputData inputData = new InputData();
         List<UserDetails> userDetailsList = null;
-        List<AdDetails> adDetailsList= null;
+        List<AdDetails> adDetailsList = null;
+
+        inputData.setClientType(role.name());
 
-        inputData.setClientType(clientType);
-        if(MERCHANT.equals(clientType)) {
-            userDetailsList = new ArrayList<>(args.length);
-            for (int i =2; i< args.length; i++) {
-                UserDetails userDetails = new UserDetails(args[i]);
+        if (Role.MERCHANT.name().equals(role.name())) {
```

Comment:
`role == Role.MERCHANT` 

This is how enums are used. Likewise for below.

Reply:
done